### ISSUE TITLE
DataTable: No objects to concatenate

### ISSUE THREAD

**Issue by paulhighams** [datapane/datapane#120]
### System Information
- OS: Ubuntu
- Python version: 3.8
- Python environment: pip
- Using jupyter: false
- Datapane version: using online community version, cant find the version number

### Bug / Issue

Pandas Dataframe generated from a query to neo4j graph database. 7 columns about 100 rows
When converted to a data pane Table it works fine
When converted to a data oane DataTable crashes, stacktrace below

Code

	query_stringfb = '''
	MATCH (cntry:Country)<-[r:REPRESENTS]-(ath:Athlete {Sex: "Female"})-[att:ATTENDS]->(cmp:Competition)<-[cs:CONSISTS_OF]-(ev:Event)-[:IDENTIFIED_BY]->(e:EventType {EventTypeName: "World Cup"})
	MATCH (cmp:Competition)<-[cl:CLASSIFIES]-(ct:CompType)
	WHERE ct.CompTypeName = "Boulder"
	RETURN ath.PersonName as Athlete, COUNT(att.FinOne) as First, COUNT(att.FinTwo) AS Second,
	 COUNT(att.FinThree) AS Third, COUNT(att.FinishPosition) AS Entered,(Count(att.FinOne) *100 / COUNT(att.FinishPosition)) AS WinPercent,
	 ((Count(att.FinOne)+Count(att.FinTwo)+Count(att.FinThree)) *100 / COUNT(att.FinishPosition)) AS PodiumPercent
	ORDER BY WinPercent DESC, First DESC, Second DESC, Third DESC
	'''
	#run the query and put the answer in a dataframe
	dtf_datafb = DataFrame([dict(_) for _ in conn.query(query_stringfb)])
	#discard rows where they have never podiumed
	dtfb = dtf_datafb[(dtf_datafb.First != 0) | (dtf_datafb.Second != 0) | (dtf_datafb.Third != 0)]
	dtfb.rename(columns = {'WinPercent':'Win %', 'PodiumPercent':'Podium %'}, inplace = True)
	#produce a datpane table
	report = dp.Report(
			dp.Page(
				dp.Select(blocks=[
					**dp.DataTable(dtfb, label='Female'),**
					dp.Table(dtmb, label='Male'),
					dp.Plot (pic3, label='Wins by athlete by country'),
					dp.Text (bnews, label='News')
				]),
				title="Boulder"
			),
			dp.Page(
				dp.Select(blocks=[
					dp.Table(dtfl, label='Female'),
					dp.Table(dtml, label='Male'),
					dp.Plot (pic5, label='Wins by athlete by country'),
					dp.Text (lnews, label='News')
				]),
				title="Lead"
			),
			dp.Page(
				dp.Select(blocks=[
					dp.Table(dtfs, label='Female'),
					dp.Table(dtms, label='Male'),
					dp.Text (snews, label='News')
				]),
				title="Speed"
			),
		type=dp.ReportType.DASHBOARD
		)
	report.publish(name='IFSC World Cup GOAT Stats', open=True)

Stacktrace:

Traceback (most recent call last):
  File "connect-neo4j.py", line 228, in <module>
    dp.DataTable(dtfb, label='Female'),
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/datapane/client/api/report/blocks.py", line 720, in __init__
    fn = save_df(df)
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/datapane/client/api/dp_object.py", line 216, in save_df
    ArrowFormat.save_file(fn.name, df)
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/datapane/common/datafiles.py", line 56, in save_file
    df = process_df(df)
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/datapane/common/df_processor.py", line 103, in process_df
    timedelta_to_str(df)
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/datapane/common/df_processor.py", line 44, in timedelta_to_str
    df[df_td.columns] = np.where(pd.isnull(df_td), pd.NA, df_td.astype("string"))
  File "/home/paulhighams/.local/lib/python3.8/site-packages/pandas/core/generic.py", line 5702, in astype
    result = pd.concat(results, axis=1, copy=False)
  File "/home/paulhighams/.local/lib/python3.8/site-packages/pandas/core/reshape/concat.py", line 271, in concat
    op = _Concatenator(
  File "/home/paulhighams/.local/lib/python3.8/site-packages/pandas/core/reshape/concat.py", line 329, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate
Exception ignored in: <function Driver.__del__ at 0x7f1631fdf700>
Traceback (most recent call last):
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/neo4j/__init__.py", line 279, in __del__
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/neo4j/__init__.py", line 309, in close
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/neo4j/io/__init__.py", line 506, in close
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/neo4j/io/__init__.py", line 495, in remove
  File "/home/paulhighams/anaconda3/lib/python3.8/site-packages/neo4j/io/_bolt4.py", line 439, in close
AttributeError: 'NoneType' object has no attribute 'debug'

**Comment by JobCollins** [comment]
I am also having the same issue with `datapane 0.11.8`

OS: Windows 10 Pro
Python 3.8.3

Code I'm running:

```
top_10_df = run_query(top_10_tracks_query)

dp.Report(
    dp.DataTable(top_10_df)
).publish(name="Music Sales")

```

Here is my stack trace:
```
Traceback (most recent call last):
  File "c:/Users/intel/Desktop/stuff/datapane-sql-interactive-dashboard/top_tracks.py", line 26, in <module>
    dp.DataTable(top_10_df)
  File "C:\Users\intel\anaconda\lib\site-packages\datapane\client\api\report\blocks.py", line 720, in __init__
    fn = save_df(df)
  File "C:\Users\intel\anaconda\lib\site-packages\datapane\client\api\dp_object.py", line 216, in save_df
    ArrowFormat.save_file(fn.name, df)
  File "C:\Users\intel\anaconda\lib\site-packages\datapane\common\datafiles.py", line 56, in save_file
    df = process_df(df)
  File "C:\Users\intel\anaconda\lib\site-packages\datapane\common\df_processor.py", line 103, in process_df
    timedelta_to_str(df)
  File "C:\Users\intel\anaconda\lib\site-packages\datapane\common\df_processor.py", line 44, in timedelta_to_str
    df[df_td.columns] = np.where(pd.isnull(df_td), pd.NA, df_td.astype("string"))
  File "C:\Users\intel\anaconda\lib\site-packages\pandas\core\generic.py", line 5702, in astype
    result = pd.concat(results, axis=1, copy=False)
  File "C:\Users\intel\anaconda\lib\site-packages\pandas\core\reshape\concat.py", line 271, in concat
    op = _Concatenator(
  File "C:\Users\intel\anaconda\lib\site-packages\pandas\core\reshape\concat.py", line 329, in __init__
    raise ValueError("No objects to concatenate")
ValueError: No objects to concatenate
```

**Comment by johnmicahreid** [comment]
Hi @paulhighams @JobCollins thanks for bringing this up. We're investigating the issue and should have a fix out in the next couple of weeks.

**Comment by mands** [comment]
Thanks for raising - this is fixed in version `0.11.10`
